### PR TITLE
[Host.AzureServiceBus] #349 Correlation filters

### PR DESCRIPTION
**Pull request by EtherZa**
[Correlation filters](https://learn.microsoft.com/en-us/azure/service-bus-messaging/topic-filters#correlation-filters) are more efficient then SQL filters and are the preferred way to implement filtering when the features of SQL rule filters are not required.

```c#
mbb.Consume<SomeMessage>(x => x
      .Topic("some-topic")
      .WithConsumer<SomeMessageConsumer>()
      .SubscriptionName("some-service")
      .SubscriptionCorrelationFilter(correlationId: "some-correlation-id", contentType: "application/json")
      .CreateTopicOptions((options) =>
      {
         options.RequiresDuplicateDetection = false;
      })
   );
```

### CODE REVIEW

File: docs/provider_azure_servicebus.md [hunk 1]
```diff
@@ -408,7 +408,7 @@ mbb.Consume<SomeMessage>(x => x
       .Topic("some-topic")
       .WithConsumer<SomeMessageConsumer>()
       .SubscriptionName("some-service")
-      .SubscriptionSqlFilter("1=1") // this will create a rule with SQL filter
+      .SubscriptionCorrelationFilter(correlationId: "Sample") // this will create a rule to filter messages with a CorrelationId of 'Sample'
```
Comment (zarusz) [inline docs/provider_azure_servicebus.md:411]: Would you be able to reference the subscription filter MSDN docs somewhere here or in the extension method docs?

**Comment by zarusz** [comment]
@EtherZa Do you need this in v2 (current master)? 
I would love to put new features in release/v3 branch for the upcoming release.

Good job on the feature!

File: docs/provider_azure_servicebus.md (hunk 1 above)
Comment (EtherZa) [reply docs/provider_azure_servicebus.md]: Added.

**Comment by EtherZa** [comment]
> @EtherZa Do you need this in v2 (current master)? I would love to put new features in release/v3 branch for the upcoming release.
> 
> Good job on the feature!

Thanks! It's not something that I am in desperate need of right now -I have rebased the change on release/v3.

Please let me know if would like anything else tweaked.

**Comment by EtherZa** [comment]
@zarusz It looks like there may be a missing/misconfigured secret in the build pipeline for Amazon SQS.

**Pull request merged**